### ISSUE TITLE
Lint all JavaScript in markdown files by default

### ISSUE THREAD

**Issue by feross** [standard/standard#1355]
ESLint supports the notion of "processors" which allows us to extract JS from markdown files and lint it to ensure it is in standard style. https://eslint.org/docs/user-guide/configuring#specifying-processor

That would let us eliminate this whole section of the readme which describes how to install `eslint-plugin-markdown`, run `standard` with the `--plugin` flag. In the end you need to actually run standard twice if you do this: `standard && standard --plugin markdown '**/*.md'`

I think we can do better. Let's just do the right thing out of the box.

We should probably do the same for <script> tags in HTML files too, while we're at it.

We can provide a `--no-md` or `--no-html` escape hatch for those who need it. But ideally, this just does the right thing 99.9% of the time.

**Comment by mightyiam** [comment]
Brilliant. I'd love that. Would probably break some projects.

Would that be supported merely by extending from `eslint-config-standard` or would it require the executable? I would prefer the former, if possible.

**Comment by feross** [comment]
We'd put it directly in `eslint-config-standard`.

**Comment by vweevers** [comment]
> We should probably do the same for <script> tags in HTML files too, while we're at it.

One requirement for this is that generated HTML (like a code coverage report) is excluded. Such HTML is typically git-ignored though, which tells `standard` to ignore it too, so it's probably fine.

**Comment by vweevers** [comment]
I've seen and written code examples in READMEs that are not 100% correct, for example to quickly demonstrate two ways to achieve the same thing:

```js
const x = doItOneWay()

// Alternatively:
const x = doItAnotherWay()
```

I guess `standard` would implicitly impose a rule to split such snippets? In this case, to not reassign `x`.

**Comment by feross** [comment]
@vweevers Yeah, we'd certainly need to disable some of the rules for examples, like unused variables for example. I think we can look at what `standard-markdown` disables as a starting point: https://github.com/zeke/standard-markdown/blob/2f044f48eed5f7162147e7927b65a61387200341/lib/linter.js#L8-L15

**Comment by feross** [comment]
I attempted to get this working for `standard` 16 but ran into issues. Help would be appreciated to figure it out!

Here's what I tried:

## eslint-config-standard

1. Install `eslint-plugin-markdown` as a peer dep of `eslint-config-standard` (2.0.0-rc.0 was latest at the time)
2. Add `"extends": ["plugin:markdown/recommended"]` to the `eslint.json` file.

## standard

1. Install `eslint-plugin-markdown` as a dep of `eslint-config-standard`
2. Add `extensions: ['.md']` to `eslintConfig` in `options.js`:

<img width="526" alt="Screen Shot 2020-10-28 at 3 50 55 PM" src="https://user-images.githubusercontent.com/121766/97515583-943b7480-194e-11eb-8dc8-725df3af7011.png">

## observed behavior

```
  /Users/feross/code/standard/CHANGELOG.md:654:10: 'foo' is defined but never used. (no-unused-vars)
```

Rules like `no-unused-vars` are being enforced, which should be ignored for markdown files. From the docs for `eslint-plugin-markdown`:

> Some rules that catch mistakes in regular code are less helpful in documentation. For example, no-undef would flag variables that are declared outside of a code snippet because they aren't relevant to the example. The plugin:markdown/recommended config disables these rules in Markdown files:

https://github.com/eslint/eslint-plugin-markdown#frequently-disabled-rules

Help appreciated!

**Comment by jimmywarting** [comment]
bump. would like to see this land

**Comment by LeonxLJX** [comment]
I'd like to take this one (`Lint all JavaScript in markdown files by default`). I'll look into the root cause and follow up with a PR. (claiming via @LeonxLJX)

**Comment by LeonxLJX** [comment]
I'd like to take this one (`Lint all JavaScript in markdown files by default`). I'll look into the root cause and follow up with a PR. (claiming via @LeonxLJX)
